examples/kandinsky2_2_train/tune_decoder.py

```diff
@@ -0,0 +1,616 @@
+import argparse
+import os
+import numpy as np
+from PIL import Image, ImageOps
+from tqdm import tqdm
+import pandas as pd
+import math
+from packaging import version
+
+import torch
+import torch.nn.functional as F
+import torch.utils.checkpoint
+import accelerate
+from accelerate import Accelerator
+from accelerate.logging import get_logger
+from accelerate.state import AcceleratorState
+from accelerate.utils import ProjectConfiguration, set_seed
+from torchvision import transforms
+import transformers
+from transformers import CLIPImageProcessor, CLIPVisionModelWithProjection
+from transformers.utils import ContextManagers
+
+import diffusers
+from diffusers import DDPMScheduler, UNet2DConditionModel, VQModel
+from diffusers.optimization import get_scheduler
+from diffusers.training_utils import EMAModel
+from diffusers.utils import check_min_version, deprecate, is_wandb_available
+from diffusers.models.attention_processor import LoRAAttnProcessor
+
+if is_wandb_available():
+    import wandb
+
+logger = get_logger(__name__, log_level="INFO")    
+    
+def parse_args():
+    parser = argparse.ArgumentParser(description="Simple example of finetuning Kandinsky 2.2.")
+    parser.add_argument(
+        "--image_resolution",
+        type=int,
+        default=512,
+        required=False,
+        help="Image resolution",
+    )
+    parser.add_argument(
+        "--pretrained_kandinsky_path",
+        type=str,
+        default='kandinsky-community/kandinsky-2-2-decoder',
+        required=False,
+        help="Path to pretrained model or model identifier from huggingface.co/models.",
+    )
+    parser.add_argument(
+        "--pretrained_vae_path",
+        type=str,
+        default='kandinsky-community/kandinsky-2-2-decoder',
+        required=False,
+        help="Path to pretrained vae.",
+    )
+    parser.add_argument(
+        "--pretrained_image_encoder",
+        type=str,
+        default='kandinsky-community/kandinsky-2-2-prior',
+        required=False,
+        help="Path to pretrained image encoder.",
+    )
+    parser.add_argument(
+        "--scheduler_path",
+        type=str,
+        default='kandinsky-community/kandinsky-2-2-decoder',
+        required=False,
+        help="Path to scheduler.",
+    )
+    parser.add_argument(
+        "--image_processor_path",
+        type=str,
+        default='kandinsky-community/kandinsky-2-2-prior',
+        required=False,
+        help="Path to image_processor.",
+    )
+    parser.add_argument(
+        "--train_image_folder",
+        type=str,
+        default=None,
+        required=False,
+        help="Path to train image folder.",
+    )
+    parser.add_argument(
+        "--train_images_paths_csv",
+        type=str,
+        default=None,
+        required=False,
+        help="Path to csv with train images paths(define train_image_folder or train_images_paths_csv) with column paths.",
+    )
+    parser.add_argument(
+        "--val_image_folder",
```

Comment:
I don't think we used validation dataset? Are we planning on adding a validation process?